workflow/controller/artifact_gc.go

```diff
@@ -104,11 +104,7 @@ func (woc *wfOperationCtx) processArtifactGCStrategy(ctx context.Context, strate
 	woc.log.Debugf("processing Artifact GC Strategy %s", strategy)
 
 	// Search for artifacts
-	artifactSearchResults := woc.wf.SearchArtifacts(&wfv1.ArtifactSearchQuery{
```

Comment:
Why not just change to wow.execWF.SearchArtifacts?

Reply:
Tried that at first but the execWf doesn’t seem to include the Workflow Status (for node status). Should it?

Reply:
You're correct, you must inspect `woc.execWF.Spec` and `woc.wf.Status`. The former is what to run, the latter what to update at the end of reconciliation.

You should not query across the two.

Reply:
Does this look okay then? (I’m not sure if you’re saying you feel that I’m querying between the two)

Reply:
@juliev0 Make sure you include the test for workflowdefautls, workflow template

Reply:
@sarabala1979 I just ran a couple of tests locally to test WorkflowDefaults. In one, I just used a Workflow by itself and verified that it took the default from WorkflowDefaults. In the second, I used a WorkflowTemplate as well. As discussed, since the workflow-controller-configmap is not being dynamically reloaded, it's difficult to test this as part of the e2e test, since otherwise I need to change `test/e2e/manifests/mixins/workflow-controller-configmap.yaml`, which could have consequences for other e2e tests.